test/reply-trailers.test.js

```diff
@@ -111,29 +111,76 @@ test('send trailers when payload is json', t => {
 })
 
 test('send trailers when payload is stream', t => {
-  t.plan(7)
+  t.plan(2)
 
-  const fastify = Fastify()
+  t.test('single trailers', t => {
+    t.plan(7)
 
-  fastify.get('/', function (request, reply) {
-    reply.trailer('ETag', function (reply, payload, done) {
-      t.same(payload, null)
-      done(null, 'custom-etag')
+    const fastify = Fastify()
+
+    fastify.get('/', function (request, reply) {
+      reply.trailer('ETag', function (reply, payload, done) {
+        t.same(typeof payload.pipe === 'function', true)
+        done(null, 'custom-etag')
+      })
+      const stream = Readable.from([JSON.stringify({ hello: 'world' })])
+      reply.send(stream)
+    })
+
+    fastify.inject({
+      method: 'GET',
+      url: '/'
+    }, (error, res) => {
+      t.error(error)
+      t.equal(res.statusCode, 200)
+      t.equal(res.headers['transfer-encoding'], 'chunked')
+      t.equal(res.headers.trailer, 'etag')
+      t.equal(res.trailers.etag, 'custom-etag')
+      t.notHas(res.headers, 'content-length')
     })
-    const stream = Readable.from([JSON.stringify({ hello: 'world' })])
-    reply.send(stream)
   })
 
-  fastify.inject({
-    method: 'GET',
-    url: '/'
-  }, (error, res) => {
-    t.error(error)
-    t.equal(res.statusCode, 200)
-    t.equal(res.headers['transfer-encoding'], 'chunked')
-    t.equal(res.headers.trailer, 'etag')
-    t.equal(res.trailers.etag, 'custom-etag')
-    t.notHas(res.headers, 'content-length')
```

Comment:
Can you keep the previous test as it was? It's hard to judge if we are breaking anything.

Reply:
The test is keep. I just group them in `sub-test`.
I can rearrange the test.